graphql.go

```diff
@@ -386,6 +389,28 @@ func newError(code string, err error) Error {
 	}
 }
 
+type ErrResponseStatusNotOK struct {
+	body       io.ReadCloser
```

Comment:
We should decode and close the response body. This will cause HTTP connection leaks if other users aren't aware and explicitly close the body.

IMO this error would be:

```go
type NetworkError struct {
  message string
  statusCode int
}
```

Reply:
Moreover, [this line](https://github.com/hasura/go-graphql-client/blob/a5b093eb0610ba4072ad763f94b0017a847c7331/graphql.go#L167) closes the response body when the function ends. Returning the body doesn't work.

Reply:
Ah yeah, that's a great point.